activation/nipost.go

```diff
@@ -115,100 +123,65 @@ func NewNIPostBuilder(
 }
 
 // updatePoETProver updates poetProver reference. It should not be executed concurently with BuildNIPST.
-func (nb *NIPostBuilder) updatePoETProver(poetProver PoetProvingServiceClient) {
+func (nb *NIPostBuilder) updatePoETProvers(poetProvers []PoetProvingServiceClient) {
 	// reset the state for safety to avoid accidental erroneous wait in Phase 1.
 	nb.state = &BuilderState{
 		NIPost: &types.NIPost{},
 	}
-	nb.poetProver = poetProver
-	nb.log.With().Info("updated poet proof service client")
+	nb.poetProvers = poetProvers
+	nb.log.With().Info("updated poet proof service clients", log.Int("count", len(nb.poetProvers)))
 }
 
-// BuildNIPost uses the given challenge to build a NIPost. "atxExpired" and "stop" are channels for early termination of
+// BuildNIPost uses the given challenge to build a NIPost. "deadline" is a channel for early termination of
 // the building process. The process can take considerable time, because it includes waiting for the poet service to
 // publish a proof - a process that takes about an epoch.
-func (nb *NIPostBuilder) BuildNIPost(ctx context.Context, challenge *types.Hash32, atxExpired chan struct{}) (*types.NIPost, error) {
+func (nb *NIPostBuilder) BuildNIPost(ctx context.Context, challenge *types.Hash32, deadline chan struct{}) (*types.NIPost, time.Duration, error) {
```

Comment:
I would instead of returning the duration to create a post here, make it a field that can be read from by the user of the `NIPostBuilder`.